### PR TITLE
core(full-page-screenshot): resolve inaccuracies related to dpr

### LINKED ISSUE #11121: full-page-screenshot: handle elements beyond MAX_SCREENSHOT_HEIGHT
ref https://github.com/GoogleChrome/lighthouse/pull/10716#issuecomment-657904592 and #11118 

i made https://output.jsbin.com/koqamuh/4/quiet as a test page for us. it has a (CSS) pixel ruler running down the side.

Here's a report of the page: https://marked-cable.surge.sh/color-contrast-over16000px-koqamuh.html

The elem screenshots start failing between 6200 and 6300 px

image

---------

#### DPR stuff..

6300  is  roughly 16384 / 2.625.. so  it's to do with some DPR adjustments.. (I'm capturing this on my imacpro with a native dpr of 2.)

FWIW i tried removing the `/ dpr` in maxScreenshotHeight: 
https://github.com/GoogleChrome/lighthouse/blob/06d3e1b8d69c9b7eb0a9b94c08632b097eb1eeec/lighthouse-core/gather/gatherers/full-page-screenshot.js#L75

and after doing that, [it cut out](https://marked-cable.surge.sh/color-contrast-over16000px-koqamuh-nodprdivisor.html) at 16300 . (right before 16384, the real max)

image

that much seems good.
of course, the placement of the markers are off because of some other dpr math I didnt account for.

### PR DESCRIPTION
**Pull request by connorjclark**
Fixes #11121

### CODE REVIEW

**Review by patrickhulce — COMMENTED** [review]
DPR change itself looks fine but exposed issue with our reset?

File: lighthouse-core/gather/gatherers/full-page-screenshot.js [hunk 1]
```diff
@@ -31,7 +31,6 @@ class FullPageScreenshot extends Gatherer {
   async _takeScreenshot(passContext, maxScreenshotHeight) {
     const driver = passContext.driver;
     const metrics = await driver.sendCommand('Page.getLayoutMetrics');
-    const deviceScaleFactor = await driver.evaluateAsync('window.devicePixelRatio');
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/gather/gatherers/full-page-screenshot.js:34]: yeah that was an oversight. I'll open a separate PR for that.

**Review by patrickhulce — APPROVED** [review]

File: lighthouse-core/gather/gatherers/full-page-screenshot.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/full-page-screenshot.js:34]: alright 👍

File: lighthouse-core/gather/gatherers/full-page-screenshot.js [hunk 2]
```diff
@@ -48,7 +47,7 @@ class FullPageScreenshot extends Gatherer {
       screenHeight: height,
       width,
       screenWidth: width,
-      deviceScaleFactor,
+      deviceScaleFactor: 1,
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/full-page-screenshot.js:50]: Oh kinda felt like all bets were off on reacting to a page that's 16000px tall to me. Do you suspect this is going to be fundamentally worse? (I was also under the impression the screenshot would happen after the first render not necessarily waiting for many images to load from the network, is that not the case?)

**Comment by connorjclark** [comment]
Closing, I think we have a different problem to solve. see #11121

**Pull request closed without merging**